src/net/KNet/Specific/Serialization/KNetSerialization.cs

```diff
@@ -247,49 +248,49 @@ public static byte[] DeserializeByteArray(string topic, byte[] data)
             return data;
         }
 
-        static readonly Deserializer<ByteBuffer> _ByteBufferDeserializer = new ByteBufferDeserializer();
+        static readonly ByteBufferDeserializer _ByteBufferDeserializer = new ByteBufferDeserializer();
         /// <summary>
         /// Deserialize a <see cref="SerializationType.ByteBuffer"/>
         /// </summary>
         public static ByteBuffer DeserializeByteBuffer(string topic, byte[] data)
         {
-            return _ByteBufferDeserializer.Deserialize(topic, data);
+            return JVMBridgeBase.Wraps<ByteBuffer>(_ByteBufferDeserializer.Deserialize(topic, data) as IJavaObject);
         }
 
-        static readonly Deserializer<Bytes> _BytesDeserializer = new BytesDeserializer();
+        static readonly BytesDeserializer _BytesDeserializer = new BytesDeserializer();
         /// <summary>
         /// Deserialize a <see cref="SerializationType.Bytes"/>
         /// </summary>
         public static Bytes DeserializeBytes(string topic, byte[] data)
         {
-            return _BytesDeserializer.Deserialize(topic, data);
+            return JVMBridgeBase.Wraps< Bytes>(_BytesDeserializer.Deserialize(topic, data) as IJavaObject);
         }
 
-        static readonly Deserializer<double> _DoubleDeserializer = new DoubleDeserializer();
+        static readonly DoubleDeserializer _DoubleDeserializer = new DoubleDeserializer();
         /// <summary>
         /// Deserialize a <see cref="SerializationType.Double"/>
         /// </summary>
         public static double DeserializeDouble(string topic, byte[] data)
         {
-            return _DoubleDeserializer.Deserialize(topic, data);
+            return (double)_DoubleDeserializer.Deserialize(topic, data);
```

Comment:
The cast does not work right in this way, you must use something like: JVMBridgeBase.Wraps<Java.Lang.Double>(_DoubleDeserializer.Deserialize(topic, data) as IJavaObject);